src/sqlfluff/rules/std.py

```diff
@@ -2220,41 +2220,139 @@ def _eval(self, segment, **kwargs):
         error_buffer = []
         if segment.type == 'with_compound_statement':
-            expecting_blank_line = False
-            blank_line_found = False
-            blank_line_started = False
-            fix_point = None
-            cte_end_segments = ('end_bracket', 'comma') if self.comma_style == "trailing" else ('end_bracket')
-            for seg in segment.segments:
-                if seg.type in cte_end_segments:
-                    expecting_blank_line = True
-                    blank_line_found = False
-                    blank_line_started = False
-                elif seg.is_code:
-                    if expecting_blank_line:
-                        if not blank_line_found:
-                            fix_point = fix_point or seg
-                            fixes = [LintFix(
-                                'create', fix_point,
-                                [self.make_newline(pos_marker=fix_point.pos_marker)]
-                                # Two newlines if there isn't one at all, otherwise one.
-                                * (1 if blank_line_started else 2)
-                            )]
-                            error_buffer.append(
-                                LintResult(anchor=seg, fixes=fixes)
-                            )
-                        expecting_blank_line = False
-                        blank_line_found = False
-                        blank_line_started = False
-                        fix_point = None
-                elif seg.type == 'newline':
-                    if not blank_line_found:
-                        if blank_line_started:
-                            blank_line_found = True
+            # First we need to find all the commas, the end brackets, the
+            # things that come after that and the blank lines in between.
+
+            # Find all the closing brackets. They are our anchor points.
+            bracket_indices = []
+            for idx, seg in enumerate(segment.segments):
+                if seg.type == 'end_bracket':
+                    bracket_indices.append(idx)
+
+            # Work through each point and deal with it individually
+            for bracket_idx in bracket_indices:
+                forward_slice = segment.segments[bracket_idx:]
+                seg_idx = 1
+                line_idx = 0
+                comma_seg_idx = None
+                blank_lines = 0
+                comma_line_idx = None
+                line_blank = False
+                comma_style = None
+                line_starts = {}
+                comment_lines = []
+
+                self.logger.info(
+                    "## CTE closing bracket found at %s, idx: %s. Forward slice: %.20r",
+                    forward_slice[0].pos_marker,
+                    bracket_idx,
+                    ''.join(elem.raw for elem in forward_slice)
+                )
+
+                # Work forward to map out the following segments.
+                while True:
```

Comment:
Generally best to avoid while True unless absolutely necessary, it's easier to read and safer to put the logic around when to loop into the while condition instead of `True`

Reply:
Good catch. Bad habit die hard.